### PR TITLE
Update DNSPolicy model according to load balance change

### PR DESCRIPTION
**Pull request by jsmolar**

### CODE REVIEW

File: testsuite/kuadrant/policy/dns.py
Comment (azgabur) [inline testsuite/kuadrant/policy/dns.py:1]: This change is not enough. Loadbalancing section got refactored as well so `LoadBalancing` class needs to reflect it as well. 
In the `if load_balancing:` the `routingStrategy` should be removed too

**Review by azgabur — COMMENTED** [review]
See https://github.com/Kuadrant/kuadrant-operator/issues/855

File: testsuite/kuadrant/policy/dns.py [hunk 1]
```diff
@@ -25,15 +26,9 @@ def _check(obj):
...
-
-    def asdict(self):
-        """Custom asdict due to nested structure."""
-        return {
-            "geo": {"defaultGeo": self.default_geo},
-            "weighted": {"defaultWeight": self.default_weight},
-        }
+    defaultGeo: bool
+    geo: str
+    weight: Optional[int]
```
Comment (averevki) [inline testsuite/kuadrant/policy/dns.py:31]: i think weight is still a required field

Comment (azgabur) [reply testsuite/kuadrant/policy/dns.py:31]: According to reference, looks like no :thinking: 
https://github.com/Kuadrant/kuadrant-operator/blob/main/doc/reference/dnspolicy.md#loadbalancingspec

Comment (averevki) [reply testsuite/kuadrant/policy/dns.py:31]: openshift dashboard says another thing. It might be wrong though

**Review by averevki — APPROVED** [review]

**Pull request merged**